core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java

```diff
@@ -67,7 +70,8 @@ public CachedConnection reserveAny() {
     }
   }
 
-  private Map<ThriftTransportKey,CachedConnections> cache = new HashMap<>();
+  private ConcurrentHashMap<ThriftTransportKey,Lock> locks = new ConcurrentHashMap<>();
+  private ConcurrentHashMap<ThriftTransportKey,CachedConnections> cache = new ConcurrentHashMap<>();
   private Map<ThriftTransportKey,Long> errorCount = new HashMap<>();
```

Comment:
These error tracking maps and sets are accessed while different locks are held.  I created lbschanno/accumulo#2 as one possible way to bring those under the same lock.

Reply:
Okay, looks good to me. I'll merge the changes in.